data_preprocess.py

```diff
@@ -37,24 +37,21 @@ def convert_dcm(image_paths, output_path="MAIDA/data", PNG=True):
 def downsize_one_image(image_path, resized_dim):
     # breakpoint()
     image = Image.open(image_path)
+
+    # Grayscale MinMax Normalization
     img = np.array(image)
-    if img.max() > 255:
-        img = (img / 256).astype("uint8")
+    img = (img - img.min()) / (img.max() - img.min()) * 255.0
+    img = img.astype('uint8')
     image = Image.fromarray(img).convert("RGB")
-    w, h = image.size
 
-    # crop
-    if w > h:
-        image = image.crop(
-            ((w - h) / 2, 0, (w + h) / 2, h)
-        )  # cut evenly from left and right
-    else:
-        image = image.crop(
-            (0, (h - w) * 0.2, w, h - (h - w) * 0.8)
-        )  # cut more from bottom and less from top
+    # Determine the desired size for the square image
+    desired_size = max(image.size)
 
+    # Pad the image to the desired size
+    padded_image = ImageOps.pad(image, (desired_size, desired_size))
```

Comment:
minor suggestion: add centering=(0.5, 0.5) to the arguments and leave a comment that the original image will be centered after padding